libvast/src/system/index.cpp

```diff
@@ -283,9 +287,14 @@ caf::error index_state::load_from_disk() {
     if (!parsers::uuid(stem.string(), partition_uuid))
       continue;
     auto ext = entry.path().extension();
-    if (ext.empty())
-      partitions.push_back(partition_uuid);
-    else if (ext == std::filesystem::path{".mdx"})
+    if (ext.empty()) {
+      if (entry.file_size() >= FLATBUFFERS_MAX_BUFFER_SIZE) {
+        auto store_path = dir / ".." / store_path_for_partition(partition_uuid);
+        if (std::filesystem::exists(store_path, err))
+          oversized_partitions.push_back(partition_uuid);
```

Comment:
What about the case where no store file exists? (ie. because VAST crashed while saving it) There will still be an oversized partition on disk, so we should probably at least issue a warning.

Reply:
In that case the data could also be stored in the archive. That is actually more likely, because the segment store should usually be quite a bit quicker with packing.

Reply:
Hm, fair enough. But if we just ignore it here we have this 2G file that we just silently ignore. Should we maybe issue a message and delete the partition?

Reply:
A message maybe, but I'm against automatic deletion in this case, because then it will be impossible to recover the data even manually.